.github/pull_request_template.md

```diff
@@ -32,6 +32,7 @@ Please check all boxes (including N/A items)
 
 - [ ] __docs__: All relevant [docs](https://github.com/aws/aws-step-functions-data-science-sdk-python/tree/main/doc) updated
 - [ ] __docstrings__: All public APIs documented
+- [ ] __Generate docs locally__: Updated docs are rendered properly (see [Contributing guide](https://github.com/aws/aws-step-functions-data-science-sdk-python/blob/main/CONTRIBUTING.md)) 
```

Comment:
question: what does "rendered properly" mean in this context? - valid links? style? etc... can we make it more prescriptive so it isn't as subjective?

aside -> we should automate this as part of build/pipeline verification. currently, we're relying on good intentions.